MantidQt/MantidWidgets/src/InstrumentView/InstrumentActor.cpp

```diff
@@ -757,9 +761,9 @@ void InstrumentActor::setupPickColors() {
 void InstrumentActor::cacheDetPos() const {
   if (m_detPos.size() != m_detIDs.size()) {
     m_detPos.clear();
-    for (size_t i = 0; i < m_detIDs.size(); i++) {
-      IDetector_const_sptr det = this->getDetector(i);
-      m_detPos.push_back(det->getPos());
+    for (size_t pickID = 0; pickID < m_detIDs.size(); pickID++) {
```

Comment:
On the other hand `pickID`s should probably distinguish detectors at different time steps, so they cannot be identical to the detector indices.